torch/distributed/pipeline/sync/pipe.py

```diff
@@ -137,64 +137,65 @@ def _split_module(modules: nn.Sequential) -> Tuple[List[nn.Sequential], List[tor
 
 class Pipe(Module):
     """Wraps an arbitrary :class:`nn.Sequential <torch.nn.Sequential>` module
-    to train on Pipe_. If the module requires lots of memory, Pipe will be
-    very efficient.
-    ::
+    to train on using synchronous pipeline parallelism. If the module requires lots of memory and doesn't fit on a single GPU, pipeline parallelism is a useful technique to employ for training.
 
-        model = nn.Sequential(a, b, c, d)
-        model = Pipe(model, chunks=8)
-        output = model(input)
+    The implementation is based on the torchgpipe_ paper.
 
-    .. _Pipe: https://arxiv.org/abs/2004.09910
+    .. _torchgpipe: https://arxiv.org/abs/2004.09910
 
     Pipe combines pipeline parallelism with checkpointing to reduce peak
     memory required to train while minimizing device under-utilization.
 
-    You should place all the modules on the appropriate devices before passing
-    them to this API and wrap them into an ``nn.Sequential`` module defining the
+    You should place all the modules on the appropriate devices and wrap them into an :class:`nn.Sequential <torch.nn.Sequential>` module defining the
     desired order of execution.
 
-    Args:
-        module (``torch.nn.Sequential``):
-            Sequential module to be parallelized using pipelining. Each module
+    Arguments:
+        module (:class:`nn.Sequential <torch.nn.Sequential>`):
+            sequential module to be parallelized using pipelining. Each module
             in the sequence has to have all of its parameters on a single
             device. Each module in the sequence has to either be an nn.Module
-            or ``nn.Sequential`` (to combine multiple sequential modules on a single
+            or :class:`nn.Sequential <torch.nn.Sequential>` (to combine multiple sequential modules on a single
             device)
-
-    Keyword Args:
         chunks (int):
             number of micro-batches (default: ``1``)
```

Comment:
This is assuming the first dimension is the batch dimension? Do we want to mention what if `chunks` > the samples in a batch?

Reply:
This clarification probably belongs in the forward method, I've updated those docs to reflect this.